apis/v1alpha2/opentelemetrycollector_types.go

```diff
@@ -165,6 +166,84 @@ type OpenTelemetryCollectorSpec struct {
 	DeploymentUpdateStrategy appsv1.DeploymentStrategy `json:"deploymentUpdateStrategy,omitempty"`
 }
 
+// TargetAllocatorEmbedded defines the configuration for the Prometheus target allocator, embedded in the
+// OpenTelemetryCollector spec.
+type TargetAllocatorEmbedded struct {
```

Comment:
should we have this use the common fields?

Reply:
I'd rather do that in the TargetAllocator CR proper, and slowly deprecate most of the common fields in this embedded struct. For the migration, I'd like to keep them the same for simplicity.